common/db/db.go

```diff
@@ -548,6 +551,27 @@ func CanIgnoreError(err error) bool {
 	return false
 }
 
+// Returns whether the tools can needs to modify a timeseries bucket to turn off mixed schema.
+func TimeseriesBucketNeedsMixedSchema(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	switch mongoErr := err.(type) {
+	case mongo.WriteError:
+		return mongoErr.Code == ErrCannotInsertTimeseriesBucketsWithMixedSchema
+
+	case mongo.BulkWriteException:
+		for _, writeErr := range mongoErr.WriteErrors {
```

Comment:
It looks like this is only checking the first error? Should we instead be returning true if any or all are 408 errors?

Reply:
Good catch!